packages/preact/inject/src/init.tsx

```diff
@@ -0,0 +1,18 @@
+import { createStore } from "@preact/common/store/store"
+
+import { InitConfig } from "./config"
+import { injectAutocomplete } from "./init/injectAutocomplete"
+import { injectCategory } from "./init/injectCategory"
+import { injectSerp } from "./init/injectSerp"
+
+export async function init(config: InitConfig) {
+  if (config.autocomplete) {
+    await injectAutocomplete()
+  }
+  if (config.category && (!config.category.isCategoryPage || config.category.isCategoryPage?.())) {
```

Comment:
here I'd say we leave the burden to determine if a category or search page should be initialized to the caller